lib/src/authentication/services/auth_service.dart

```diff
@@ -1,13 +1,80 @@
 import "package:firebase_auth/firebase_auth.dart";
+import "package:google_sign_in/google_sign_in.dart";
+import "package:kana_to_kanji/src/locator.dart";
+import "package:logger/logger.dart";
 
 /// [AuthService] Manage all the authentications processes related to firebase.
 /// It's for 2 objectives :
 /// - Isolate the code to have a better reusability
 /// - Facilitate unit tests
 class AuthService {
+  final Logger _logger = locator<Logger>();
+
+  final GoogleSignIn _googleSignIn = GoogleSignIn();
+  final AppleAuthProvider _appleAuthProvider = AppleAuthProvider();
+
+  Future<User?> getCurrentUser() async =>
+      Future.value(FirebaseAuth.instance.currentUser);
+
   /// Sign in anonymously using Firebase.
-  /// Returns a UserCredential which contains all the information of the current
-  /// user, its token...
-  Future<UserCredential> signInAnonymously() =>
-      FirebaseAuth.instance.signInAnonymously();
+  /// returns : Future<UserCredential?> - Contains all the information of the
+  /// current user, its token...
+  /// The UserCredential is null if an error occurs.
+  Future<UserCredential?> signInAnonymously() async {
+    try {
+      return await FirebaseAuth.instance.signInAnonymously();
+    } on FirebaseAuthException catch (e) {
+      _logger.e(e);
+      return Future.value();
```

Comment:
Same here, is it for the lint?

Reply:
I changed it for a `return null` as I need to return something in a case of exception